File: src/main/java/de/tuberlin/spreadalgo/OutFunction.java
package de.tuberlin.spreadalgo;

import java.util.Collection;

/**
 * The Out-Function defines how markers are passed to edges (links) if a node is activated.
 * This might include a weighting or a decision to which edges to pass markers to, or which information to add to the marker.
 */
public interface OutFunction {

	Collection<SpreadingStep> compute(Node node);

}

File: marker_passing/out_function.py
from __future__ import annotations
from abc import ABC, abstractmethod
from typing import Collection, TYPE_CHECKING

if TYPE_CHECKING:
    from .node import Node
    from .spreading_step import SpreadingStep


class OutFunction(ABC):
    """
    Defines how markers are passed to outgoing links when a node fires.
    May apply weighting, filtering, or enrichment to the markers.
    """

    @abstractmethod
    def compute(self, node: "Node") -> Collection["SpreadingStep"]:
        ...
